### PR TITLE
feat(BadgedResource): Add BadgedResource component to patternfly-react-extensions

### LINKED ISSUE #775: K8S Resource Type Badges - new component
Would like a new component to support the creation of K8S Resource Type Badges to be used in the OpenShift product portfolio.  

[This document](https://docs.google.com/document/d/1epeNjOD-BCPelgyadRoox6oL8DXN1CcQitVcHrGdZgY/edit) captures the design

### PR DESCRIPTION
**Pull request by jeff-phillips-18**
fixes #775

cc @rhamilto 

Direct storybook link: 
https://829-pr-patternfly-react-patternfly.surge.sh/patternfly-3/index.html?selectedKind=patternfly-react-extensions%2FWidgets%2FBadged%20Resource&selectedStory=BadgedResource

### CODE REVIEW

**Comment by serenamarie125** [comment]
We may want to do that next, otherwise, it will be a nightmare

File: packages/patternfly-3/patternfly-react-extensions/src/components/BadgedResource/BadgedResource.js [hunk 1]
```diff
@@ -0,0 +1,110 @@
...
+  kindStr,
+  resourceName,
+  large,
+  tipDelay,
+  href,
+  onClick,
+  ...props
+}) => {
+  const handleClick = e => {
+    if (!href) {
```
[the remark this answers is not in the document]
Comment (jschuler) [reply packages/patternfly-3/patternfly-react-extensions/src/components/BadgedResource/BadgedResource.js]: I like it, thanks for tagging me. I'll do the same over in PF4 land for the NavItem

**Comment by priley86** [comment]
it looks like aXe has some color contrast issues with the light blue and light green labels, as well as our existing link text colors. I'm not sure it if's worth changing the link text colors, but maybe we consider different shades for the badge colors? I'll leave this to the experts to determine here...

screen shot 2018-10-29 at 3 34 58 pm

**Comment by jeff-phillips-18** [comment]
@serenamarie125 Added default resourceTypes and a utility to get the color to be used for those resources (types and colors/type derived from Console)

As @priley86 mentioned above, some of those colors do not meet aXe requirements for contrast. I can leave them to remain consitent with Console or update them. Please let me know.

**Comment by serenamarie125** [comment]
I'm not sure what aXe requirements are, but these were taken from the PF color palette (or that's what we specified originally )

**Comment by jeff-phillips-18** [comment]
The colors are from the palette but that doesn't automatically make it contrast well with a white foreground. I believe these can be left 'as is' for now since they are currently in use in Console and users may be used to those colors already. Better to be consistent now and update across the board when we decide that is right across consumers.

**Comment by serenamarie125** [comment]
Agreed @jeff-phillips-18 thank you

**Comment by catrobson** [comment]
@serenamarie125 can you create an issue against PF3 (I'm assuming that's where the colors came from) to just capture that these colors were not accessible?  We might be able to tweak PF3 with this, and ensure we build upon more accessible colors moving forward.

**Comment by seanforyou23** [comment]
Wanted to chime in and +1 tweaking the base color palette to meet WCAG requirements mentioned here https://www.patternfly.org/styles/color-palette/ - this would have a huge positive impact. Would also fall in line with these related efforts in pf-next:

https://github.com/patternfly/patternfly-next/issues/620
https://github.com/patternfly/patternfly-next/issues/725

It's important for us to view the a11y requirements as a "tie breaker" when considering paths forward, where possible of course.

**Comment by jeff-phillips-18** [comment]
I'm not sure the issue is with the palette though. There are many colors in the palette that will not have good contrast with a white foreground. It's a combination thing here as all badges have a white foreground so of course some colors in the palette won't provide good contrast. This issue is with the existing use of badge colors that we are pulling from openshift/console.

**Pull request merged**